_posts/2020-06-18-a-sustainable-nuget-marketplace-will-have-to-compete-with-the-nuget-gallery.markdown

```diff
@@ -0,0 +1,368 @@
+in our IDE, where we consume most open source through the NuGet client. Which means consumers often have no idea
+there is a sponsor button.
+
+This is where https://sdkbin.com, and https://bytepack.io, and others come in. In fact, our community has been
+dreaming this up for a long, long time.
+
+Some recent calls for this:
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">An idea.<br><br>A managed NuGet[org] alternative for selling your NuGet packages.<br><br>A web interface for discovering paid-for packages. After purchase, they are made available (forever) in a private feed.<br><br>Subscriptions and payment are managed by the platform.</p>&mdash; Paul Knopf (@pauldotknopf) <a href="https://twitter.com/pauldotknopf/status/1179129145235820544?ref_src=twsrc%5Etfw">October 1, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">We basically need to transform our NuGet, Npm, VSCode plugin marketplace whatever into AppStore. <br><br>If dual licensing is not supported out of the box it will never become mainstream solution for sustaining OSS <a href="https://t.co/oRllcH0qYA">https://t.co/oRllcH0qYA</a></p>&mdash; Krzysztof Cieślak @ #BlackLivesMatter (@k_cieslak) <a href="https://twitter.com/k_cieslak/status/1222217476051742720?ref_src=twsrc%5Etfw">January 28, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+I know when I was building [MyGet](https://www.myget.org/) with [Xavier](https://twitter.com/xavierdecoster),
+we dreamt up something like a NuGet marketplace as well. Will have to check but I think I still have `nugetmarketplace.com`
+somewhere...
+
+The above is scientific proof a NuGet marketplace is needed!
+Let's look at what a NuGet marketplace should help with.
+
+### For producers: legal services
+
+Open source projects can still be open source, but sell additional services. These could be:
+
+* Dual-licensed open source
+* Paid add-on packages
+* Support services
+* Consulting services
+
+Dual-licensing will require a set of templates that projects can work with.
+If the project uses Apache-2, MIT, GPL, or other open source licenses, what will a commercial dual-license
+look like?
+
+Paid add-on packages may be open or closed source, but they will still need a license. A NuGet marketplace
+will require templates and guidance.
+
+Support services and consulting services are harder. How will a NuGet marketplace assist with these?
+Contract templates and guidance, again, may be of help here.
+
+The legal aspect will be a bit like what the open source foundations offer:
+help with protecting producers and consumers.
+
+There is one additional aspect a NuGet marketplace will have to help with: establishing rules and preventing
+their abuse. What if I have an MIT-licensed project, and someone sells it with a dual license on this
```

Comment:
The marketplaces will probably have to charge an onboarding fee to developers who want to sell on it - similar to the Apple Developer program. This helps create a higher barrier to entry for bad actors and also pays for human employees at the marketplace to verify the provenance of the developer / their work.